src/os/hostname.zig

```diff
@@ -1,11 +1,104 @@
 const std = @import("std");
+const builtin = @import("builtin");
 const posix = std.posix;
 
 pub const HostnameParsingError = error{
     NoHostnameInUri,
     NoSpaceLeft,
 };
 
+pub const UrlParsingError = std.Uri.ParseError || error{
+    HostnameIsNotMacAddress,
+    NoSchemeProvided,
+};
+
+const mac_address_length = 17;
+
+fn isUriPathSeparator(c: u8) bool {
+    return switch (c) {
+        '?', '#' => true,
+        else => false,
+    };
+}
+
+fn isValidMacAddress(mac_address: []const u8) bool {
+    // A valid mac address has 6 two-character components with 5 colons, e.g. 12:34:56:ab:cd:ef.
+    if (mac_address.len != 17) {
+        return false;
+    }
+
+    for (mac_address, 0..) |c, i| {
+        if ((i + 1) % 3 == 0) {
+            if (c != ':') {
+                return false;
+            }
+        } else {
+            if (!std.mem.containsAtLeastScalar(u8, "0123456789ABCDEFabcdef", 1, c)) {
```

Comment:
`else if` ?

Reply:
omg yeah, nice catch. fixed in 31e386afa6e4874be266ea083c773ffac9e6168e